### PR TITLE
Ee 16064 trigger archive endpoint

### PR DESCRIPTION
**Pull request by mikehalmaho**
Add an endpoint to trigger an archiving run.

### CODE REVIEW

File: src/main/java/uk/gov/digital/ho/proving/income/audit/ArchiveResource.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        AuditArchiveService auditArchiveService,
+        RequestData requestData
+    ) {
+        this.auditArchiveService = auditArchiveService;
+        this.requestData = requestData;
+    }
+
+    @PostMapping(value = "/archive")
+    public void archive() {
+        log.info("Request received on /archive");
```
Comment (owainrobertsbjss) [inline src/main/java/uk/gov/digital/ho/proving/income/audit/ArchiveResource.java:28]: We should have an Event on all our logs

File: src/test/java/uk/gov/digital/ho/proving/income/audit/ArchiveResourceTest.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        verify(mockAppender).doAppend(argThat(arg -> {
+                LoggingEvent loggingEvent = (LoggingEvent) arg;
+                return loggingEvent.getFormattedMessage().toLowerCase().contains("request received")
+                    && loggingEvent.getLevel().equals(Level.INFO);
+        }));
+    }
+
+    @Test
+    public void archive_logsResponseReturned() {
+        LogCapturer<ArchiveResource> logCapturer = LogCapturer.forClass(ArchiveResource.class);
```
Comment (owainrobertsbjss) [inline src/test/java/uk/gov/digital/ho/proving/income/audit/ArchiveResourceTest.java]: Not being used

**Review by owainrobertsbjss — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/digital/ho/proving/income/audit/ArchiveResource.java (hunk 1 above)
Comment (mikehalmaho) [reply src/main/java/uk/gov/digital/ho/proving/income/audit/ArchiveResource.java:28]: It's on the backlog for this service... but not done yet.

File: src/test/java/uk/gov/digital/ho/proving/income/audit/ArchiveResourceTest.java (hunk 2 above)
Comment (mikehalmaho) [reply src/test/java/uk/gov/digital/ho/proving/income/audit/ArchiveResourceTest.java]: :+1:

**Review by owainrobertsbjss — APPROVED** [review]

**Pull request merged**